oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -849,18 +834,21 @@ impl Runtime {
                 })?;
             }
             let statuses = self.readers_statuses(node_id, &readers);
+            // Transcribe the status for valid channels back to the original position
```

Comment:
I only had a quick look, but it seems that the logic above is still just ignoring all initially invalid channels?

Reply:
I think `readers.len() != read_handles.len()` in line `851` checks this condition and returns immediately. It only waits incorrectly if the input vector is empty! I have fixed it now.

Reply:
Ah yes you are right, though the logic seems unnecessarily convoluted now. Perhaps something to try and simplify in a separate PR?